src/lib.rs

```diff
@@ -61,34 +64,74 @@ pub mod pallet {
     #[pallet::storage]
     pub type VMStorage<T> = StorageMap<_, Blake2_128Concat, Vec<u8>, Vec<u8>>;
 
+    /// Storage for multi-signature/signer requests.
+    #[pallet::storage]
+    pub type Multisigs<T: Config> = StorageDoubleMap<
+        _,
+        Twox64Concat,
+        T::AccountId,
+        Blake2_128Concat,
+        CallHash,
+        Multisig<BlockNumberFor<T>, T::AccountId, T::MaxSignatories>,
+    >;
+
     /// MoveVM pallet configuration trait
     #[pallet::config]
     pub trait Config: frame_system::Config {
+        /// The currency mechanism.
+        type Currency: Currency<Self::AccountId>
+            + ReservableCurrency<Self::AccountId>
+            + LockableCurrency<Self::AccountId>;
+
+        /// Maximum life time for requests.
+        #[pallet::constant]
+        type MaxRequestLifetime: Get<u32>;
```

Comment:
`#[pallet::constant]`
Does this imply that the other types can be configured during the runtime?

If yes, then maybe this can also be configurable - so someone can lower it in case they start getting way too many multi-signature scripts.

Reply:
In this style, the blockchain developer can define it as a constant; this is appropriate. But if we like, we can also design it as configurable via the runtime. That's a bit extra stuff, but no problem! If we want this, I would define some storage for saving the pallet's settings / configuration and add another extrinsic call, where the sudo user can configure these properties.

Reply:
Hmm, it sounds like too much work for the current M3 - maybe we can implement it in the end - but only if you want to provide such a functionality. I don't mind either way. Till then, let's keep it simple, let's keep it constant. TIL something new 🚀

Reply:
In the past, I implemented both variants. In this case, I think the constant is a good choice :+1: